lib/ansible/module_utils/urls.py

```diff
@@ -832,30 +844,34 @@ def fetch_url(module, url, data=None, headers=None, method=None,
     r = None
     info = dict(url=url)
     try:
-        r = open_url(url, data=data, headers=headers, method=method,
-                     use_proxy=use_proxy, force=force, last_mod_time=last_mod_time, timeout=timeout,
-                     validate_certs=validate_certs, url_username=username,
-                     url_password=password, http_agent=http_agent, force_basic_auth=force_basic_auth,
-                     follow_redirects=follow_redirects)
-        info.update(r.info())
-        info['url'] = r.geturl()  # The URL goes in too, because of redirects.
-        info.update(dict(msg="OK (%s bytes)" % r.headers.get('Content-Length', 'unknown'), status=200))
-    except NoSSLError, e:
+        try:
+            r = open_url(url, data=data, headers=headers, method=method,
+                         use_proxy=use_proxy, force=force, last_mod_time=last_mod_time, timeout=timeout,
+                         validate_certs=validate_certs, url_username=username,
+                         url_password=password, http_agent=http_agent, force_basic_auth=force_basic_auth,
+                         follow_redirects=follow_redirects)
+            info.update(r.info())
+            info['url'] = r.geturl()  # The URL goes in too, because of redirects.
+            info.update(dict(msg="OK (%s bytes)" % r.headers.get('Content-Length', 'unknown'), status=200))
+        except:
+            e = sys.exc_info()[1]
+            raise
+    except NoSSLError:
         distribution = get_distribution()
         if distribution is not None and distribution.lower() == 'redhat':
             module.fail_json(msg='%s. You can also install python-ssl from EPEL' % str(e))
         else:
             module.fail_json(msg='%s' % str(e))
-    except (ConnectionError, ValueError), e:
+    except (ConnectionError, ValueError):
         module.fail_json(msg=str(e))
-    except urllib2.HTTPError, e:
+    except HTTPError:
         info.update(dict(msg=str(e), status=e.code, **e.info()))
```

Comment:
if I remember python scoping rules correctly, e from the inner try: except will be present here.   I thought this was a bit tricky/magical when I saw it but I have to admit that for this particular structure it is pretty convenient.

We can't use ```Exception as e``` style code in this file because the file also has to be usable on python2.4.  So if this does get rid of the magicalness of the inner try: except, it will have to use  ```e = sys.exc_info()[1] ```